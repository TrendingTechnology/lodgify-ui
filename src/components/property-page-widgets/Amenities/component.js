import React from 'react';
import PropTypes from 'prop-types';

import { withResponsive } from 'utils/with-responsive';
import { getUniqueKey } from 'utils/get-unique-key';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { Heading } from 'typography/Heading';
import { Link } from 'elements/Link';
import { Icon } from 'elements/Icon';
import { Modal } from 'elements/Modal';
import { Divider } from 'elements/Divider';

import { getDefaultItems } from './utils/getDefaultItems';
import { hasExtraItems } from './utils/hasExtraItems';

/**
 * The standard widget for displaying the amenities of a property.
 * @returns {Object}
 */
const Component = ({ amenities, isUserOnMobile }) => (
  <Grid>
    <GridColumn width={12}>
      <Heading>Amenities</Heading>
    </GridColumn>
    {getDefaultItems(amenities, isUserOnMobile).map(
      ({ iconName, isDisabled, label }, index) => (
        <GridColumn
          key={getUniqueKey(label, index)}
          computer={4}
          tablet={4}
          mobile={6}
        >
          <Icon isDisabled={isDisabled} label={label} name={iconName} />
        </GridColumn>
      )
    )}
    {hasExtraItems(amenities, isUserOnMobile) && (
      <GridColumn width={12}>
        <Modal trigger={<Link>View more</Link>}>
          {amenities.map(({ iconName, isDisabled, label }, index) => (
            <div key={getUniqueKey(label, index)}>
              {!!index && <Divider hasLine />}
              <Icon isDisabled={isDisabled} label={label} name={iconName} />
            </div>
          ))}
        </Modal>
      </GridColumn>
    )}
  </Grid>
);

Component.displayName = 'Amenities';

Component.propTypes = {
  /** The amenities to display as icons. */
  amenities: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * The name of the icon to display.
       * [See Semantic UI for the full list.](https://react.semantic-ui.com/elements/Icon)
       */
      iconName: PropTypes.string.isRequired,
      /** Is the amenity disabled. */
      isDisabled: PropTypes.bool,
      /** A visible label to display for the amenity. */
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  isUserOnMobile: PropTypes.bool.isRequired,
};

export const ComponentWithResponsive = withResponsive(Component);
