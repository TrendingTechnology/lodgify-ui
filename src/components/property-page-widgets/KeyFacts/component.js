import React from 'react';
import PropTypes from 'prop-types';

import { getUniqueKey } from 'lib/get-unique-key';
import { Heading } from 'typography/Heading';
import { Grid } from 'layout/Grid';
import { GridColumn } from 'layout/GridColumn';
import { IconCard } from 'elements/IconCard';

/**
 * The standard widget for displaying key facts about a property.
 * @returns {Object}
 */
export const Component = ({ keyFacts }) => (
  <Grid>
    <GridColumn width={12}>
      <Heading>Key facts</Heading>
    </GridColumn>
    <GridColumn width={12}>
      <Grid>
        {keyFacts.map(({ iconName, isDisabled, label }, index) => (
          <GridColumn
            computer={2}
            tablet={4}
            mobile={4}
            streched
            key={getUniqueKey(label, index)}
          >
            <IconCard
              isDisabled={isDisabled}
              isFilled
              label={label}
              name={iconName}
            />
          </GridColumn>
        ))}
      </Grid>
    </GridColumn>
  </Grid>
);

Component.displayName = 'KeyFacts';

Component.propTypes = {
  /** The key facts to display as icon cards. */
  keyFacts: PropTypes.arrayOf(
    PropTypes.shape({
      /**
       * The name of the icon to display.
       * [See Semantic UI for the full list.](https://react.semantic-ui.com/elements/IconCard)
       */
      iconName: PropTypes.string.isRequired,
      /** Is the key fact disabled. */
      isDisabled: PropTypes.bool,
      /** A visible label to display for the key fact. */
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
};
