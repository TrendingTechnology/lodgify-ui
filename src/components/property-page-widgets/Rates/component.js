import React from 'react';
import PropTypes from 'prop-types';

import { Dropdown } from 'inputs/Dropdown';
import { Table } from 'collections/Table';

import { getRoomTypeDropdownMarkup } from './utils/getRoomTypeDropdownMarkup';
import { getRateCategoryHeadingMarkup } from './utils/getRateCategoryHeadingMarkup';

/**
 * The standard widget for displaying the rates of a property.
 * @returns {Object}
 */
export const Component = ({
  currencyOptions,
  onChangeCurrency,
  onChangeRoomType,
  rateCategories,
  rateHeadings,
  roomTypes,
}) => (
  <section>
    {roomTypes && getRoomTypeDropdownMarkup(roomTypes, onChangeRoomType)}
    <Table
      tableHeadings={[
        <Dropdown onChange={onChangeCurrency} options={currencyOptions} />,
        ...rateHeadings,
      ]}
      tableBody={rateCategories.map(rateCategory => [
        getRateCategoryHeadingMarkup(rateCategory),
        ...rateCategory.rates,
      ])}
    />
  </section>
);

Component.defaultProps = {
  onChangeCurrency: Function.prototype,
  onChangeRoomType: Function.prototype,
  roomTypes: null,
};

Component.displayName = 'Rates';

Component.propTypes = {
  /** The currency options which the user can select. */
  currencyOptions: PropTypes.arrayOf(
    PropTypes.shape({
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.any,
    })
  ).isRequired,
  /** A function called when the currency type value changes. */
  onChangeCurrency: PropTypes.func,
  /** A function called when the room type value changes. */
  onChangeRoomType: PropTypes.func,
  /** The data that populates the table rows */
  rateCategories: PropTypes.arrayOf(
    PropTypes.shape({
      /** The name of the rate category */
      name: PropTypes.string,
      /** The date range the rate category applies to */
      dateRange: PropTypes.string,
      /** The number of guests the rate category applies to */
      numberOfGuests: PropTypes.string,
      /** The rate category cost per additional guest  */
      costPerExtraGuest: PropTypes.string,
      /** The rate category rates */
      rates: PropTypes.arrayOf(PropTypes.string),
    })
  ).isRequired,
  /** The headings for each column of the table */
  rateHeadings: PropTypes.arrayOf(PropTypes.string).isRequired,
  /** The room type options which the user can select. */
  roomTypes: PropTypes.arrayOf(
    PropTypes.shape({
      /** The source url for the image to display with the option. */
      image: PropTypes.string,
      /** The visible text for the option. */
      text: PropTypes.string.isRequired,
      /** The underlying value for the option. */
      value: PropTypes.any,
    })
  ),
};
