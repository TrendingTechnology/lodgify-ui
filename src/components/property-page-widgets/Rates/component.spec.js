import React from 'react';
import { shallow } from 'enzyme';
import {
  expectComponentToBe,
  expectComponentToHaveChildren,
  expectComponentToHaveProps,
} from '@lodgify/enzyme-jest-expect-helpers';

import { Dropdown } from 'inputs/Dropdown';
import { Grid } from 'layout/Grid';
import { Table } from 'collections/Table';

import { Component as Rates } from './component';
import {
  currencyOptions,
  expectedTableBody,
  rateCategories,
  rateHeadings,
  roomTypeOnChange,
  roomTypes,
} from './mock-data/options';

const expectedHeadings = [
  <Dropdown options={currencyOptions} />,
  ...rateHeadings,
];

const getRatesWidget = props =>
  shallow(
    <Rates
      currencyOptions={currencyOptions}
      rateCategories={rateCategories}
      rateHeadings={rateHeadings}
      {...props}
    />
  );

describe('<Rates />', () => {
  it('should render a single `section` element', () => {
    const wrapper = getRatesWidget();
    expectComponentToBe(wrapper, 'section');
  });

  describe('the `section` element', () => {
    it('should render the right children', () => {
      const wrapper = getRatesWidget();
      expectComponentToHaveChildren(wrapper, Table);
    });

    describe('if `props.roomTypes` is passed', () => {
      it('should render a `Grid` child', () => {
        const wrapper = getRatesWidget({ roomTypes, roomTypeOnChange });
        expectComponentToHaveChildren(wrapper, Grid, Table);
      });
    });
  });

  describe('the `Table` component', () => {
    it('should have the right props', () => {
      const wrapper = getRatesWidget().find(Table);
      expectComponentToHaveProps(wrapper, {
        tableHeadings: expectedHeadings,
        tableBody: expectedTableBody,
      });
    });
  });
});
