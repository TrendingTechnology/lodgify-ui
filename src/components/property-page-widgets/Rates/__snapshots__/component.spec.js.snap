// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Rates /> by default should have the right structure 1`] = `
<Fragment>
  <Grid
    areColumnsCentered={false}
    padded={true}
  >
    <ShowOnMobile
      parent="div"
      parentProps={Object {}}
    >
      <Dropdown
        error={false}
        icon={null}
        isDisabled={false}
        isValid={false}
        label=""
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        options={
          Array [
            Object {
              "text": "EUR €",
              "value": "eur",
            },
            Object {
              "text": "GBP £",
              "value": "gbp",
            },
            Object {
              "text": "USD $",
              "value": "usd",
            },
          ]
        }
        willOpenAbove={false}
      />
      <Card
        fluid={true}
        key="Season 10"
      >
        <CardContent>
          <Grid
            areColumnsCentered={false}
            padded={true}
          >
            <GridRow
              horizontalAlignContent="left"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <div>
                  <Paragraph
                    size="medium"
                    weight="heavy"
                  >
                    Season 1
                  </Paragraph>
                  <Paragraph
                    size="medium"
                    weight="light"
                  >
                    01/05/2018 - 01/08/2018
                    <br />
                    <Icon
                      color={null}
                      hasBorder={false}
                      isCircular={false}
                      isColorInverted={false}
                      isDisabled={false}
                      isLabelLeft={false}
                      labelText={null}
                      labelWeight={null}
                      name="guests"
                      path={null}
                      size={null}
                    />
                    2
                    <br />
                    1€ / Price per extra per.
                  </Paragraph>
                </div>
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 140€0"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Mon-Thu
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 130€1"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Fri
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                30€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 140€2"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sat
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 140€3"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sun
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>
      </Card>
      <Card
        fluid={true}
        key="Season 21"
      >
        <CardContent>
          <Grid
            areColumnsCentered={false}
            padded={true}
          >
            <GridRow
              horizontalAlignContent="left"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <div>
                  <Paragraph
                    size="medium"
                    weight="heavy"
                  >
                    Season 2
                  </Paragraph>
                  <Paragraph
                    size="medium"
                    weight="light"
                  >
                    01/02/2018 - 01/08/2018
                    <br />
                    <Icon
                      color={null}
                      hasBorder={false}
                      isCircular={false}
                      isColorInverted={false}
                      isDisabled={false}
                      isLabelLeft={false}
                      labelText={null}
                      labelWeight={null}
                      name="guests"
                      path={null}
                      size={null}
                    />
                    3
                    <br />
                    2€ / Price per extra per.
                  </Paragraph>
                </div>
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 210€0"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Mon-Thu
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                10€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 220€1"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Fri
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                20€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 230€2"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sat
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                30€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 240€3"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sun
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>
      </Card>
      <Card
        fluid={true}
        key="Season 32"
      >
        <CardContent>
          <Grid
            areColumnsCentered={false}
            padded={true}
          >
            <GridRow
              horizontalAlignContent="left"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <div>
                  <Paragraph
                    size="medium"
                    weight="heavy"
                  >
                    Season 3
                  </Paragraph>
                  <Paragraph
                    size="medium"
                    weight="light"
                  >
                    01/03/2018 - 01/08/2018
                    <br />
                    <Icon
                      color={null}
                      hasBorder={false}
                      isCircular={false}
                      isColorInverted={false}
                      isDisabled={false}
                      isLabelLeft={false}
                      labelText={null}
                      labelWeight={null}
                      name="guests"
                      path={null}
                      size={null}
                    />
                    4
                    <br />
                    3€ / Price per extra per.
                  </Paragraph>
                </div>
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 340€0"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Mon-Thu
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 330€1"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Fri
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                30€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 320€2"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sat
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                20€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 310€3"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sun
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                10€
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>
      </Card>
    </ShowOnMobile>
  </Grid>
  <ShowOnDesktop
    parent="div"
    parentProps={Object {}}
  >
    <Table
      className=""
      tableBody={
        Array [
          Array [
            <div>
              <Paragraph
                size="medium"
                weight="heavy"
              >
                Season 1
              </Paragraph>
              <Paragraph
                size="medium"
                weight="light"
              >
                01/05/2018 - 01/08/2018
                <br />
                <Icon
                  color={null}
                  hasBorder={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="guests"
                  path={null}
                  size={null}
                />
                2
                <br />
                1€ / Price per extra per.
              </Paragraph>
            </div>,
            "40€",
            "30€",
            "40€",
            "40€",
          ],
          Array [
            <div>
              <Paragraph
                size="medium"
                weight="heavy"
              >
                Season 2
              </Paragraph>
              <Paragraph
                size="medium"
                weight="light"
              >
                01/02/2018 - 01/08/2018
                <br />
                <Icon
                  color={null}
                  hasBorder={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="guests"
                  path={null}
                  size={null}
                />
                3
                <br />
                2€ / Price per extra per.
              </Paragraph>
            </div>,
            "10€",
            "20€",
            "30€",
            "40€",
          ],
          Array [
            <div>
              <Paragraph
                size="medium"
                weight="heavy"
              >
                Season 3
              </Paragraph>
              <Paragraph
                size="medium"
                weight="light"
              >
                01/03/2018 - 01/08/2018
                <br />
                <Icon
                  color={null}
                  hasBorder={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="guests"
                  path={null}
                  size={null}
                />
                4
                <br />
                3€ / Price per extra per.
              </Paragraph>
            </div>,
            "40€",
            "30€",
            "20€",
            "10€",
          ],
        ]
      }
      tableHeadings={
        Array [
          <Dropdown
            error={false}
            icon={null}
            isDisabled={false}
            isValid={false}
            label=""
            name=""
            onBlur={[Function]}
            onChange={[Function]}
            options={
              Array [
                Object {
                  "text": "EUR €",
                  "value": "eur",
                },
                Object {
                  "text": "GBP £",
                  "value": "gbp",
                },
                Object {
                  "text": "USD $",
                  "value": "usd",
                },
              ]
            }
            willOpenAbove={false}
          />,
          "Mon-Thu",
          "Fri",
          "Sat",
          "Sun",
        ]
      }
      tableId=""
    />
  </ShowOnDesktop>
</Fragment>
`;

exports[`<Rates /> if \`props.roomTypes\` is passed should have the right structure 1`] = `
<Fragment>
  <Grid
    areColumnsCentered={false}
    padded={true}
  >
    <GridRow
      horizontalAlignContent="left"
      verticalAlign="middle"
    >
      <GridColumn
        computer={4}
        mobile={12}
        verticalAlignContent="top"
        width={null}
      >
        <Paragraph
          size="medium"
          weight="heavy"
        >
          View Rate Information for: 
        </Paragraph>
      </GridColumn>
      <GridColumn
        computer={4}
        mobile={12}
        verticalAlignContent="top"
        width={null}
      >
        <Dropdown
          error={false}
          icon={null}
          isDisabled={false}
          isValid={false}
          label=""
          name=""
          onBlur={[Function]}
          onChange={[Function]}
          options={
            Array [
              Object {
                "image": "🎅",
                "text": "🎅",
                "value": "🎅",
              },
              Object {
                "image": "💪",
                "text": "💪",
                "value": "💪",
              },
              Object {
                "image": "👃",
                "text": "👃",
                "value": "👃",
              },
              Object {
                "image": "🧖️",
                "text": "🧖️",
                "value": "🧖️",
              },
            ]
          }
          willOpenAbove={false}
        />
      </GridColumn>
    </GridRow>
    <ShowOnMobile
      parent="div"
      parentProps={Object {}}
    >
      <Dropdown
        error={false}
        icon={null}
        isDisabled={false}
        isValid={false}
        label=""
        name=""
        onBlur={[Function]}
        onChange={[Function]}
        options={
          Array [
            Object {
              "text": "EUR €",
              "value": "eur",
            },
            Object {
              "text": "GBP £",
              "value": "gbp",
            },
            Object {
              "text": "USD $",
              "value": "usd",
            },
          ]
        }
        willOpenAbove={false}
      />
      <Card
        fluid={true}
        key="Season 10"
      >
        <CardContent>
          <Grid
            areColumnsCentered={false}
            padded={true}
          >
            <GridRow
              horizontalAlignContent="left"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <div>
                  <Paragraph
                    size="medium"
                    weight="heavy"
                  >
                    Season 1
                  </Paragraph>
                  <Paragraph
                    size="medium"
                    weight="light"
                  >
                    01/05/2018 - 01/08/2018
                    <br />
                    <Icon
                      color={null}
                      hasBorder={false}
                      isCircular={false}
                      isColorInverted={false}
                      isDisabled={false}
                      isLabelLeft={false}
                      labelText={null}
                      labelWeight={null}
                      name="guests"
                      path={null}
                      size={null}
                    />
                    2
                    <br />
                    1€ / Price per extra per.
                  </Paragraph>
                </div>
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 140€0"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Mon-Thu
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 130€1"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Fri
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                30€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 140€2"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sat
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 140€3"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sun
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>
      </Card>
      <Card
        fluid={true}
        key="Season 21"
      >
        <CardContent>
          <Grid
            areColumnsCentered={false}
            padded={true}
          >
            <GridRow
              horizontalAlignContent="left"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <div>
                  <Paragraph
                    size="medium"
                    weight="heavy"
                  >
                    Season 2
                  </Paragraph>
                  <Paragraph
                    size="medium"
                    weight="light"
                  >
                    01/02/2018 - 01/08/2018
                    <br />
                    <Icon
                      color={null}
                      hasBorder={false}
                      isCircular={false}
                      isColorInverted={false}
                      isDisabled={false}
                      isLabelLeft={false}
                      labelText={null}
                      labelWeight={null}
                      name="guests"
                      path={null}
                      size={null}
                    />
                    3
                    <br />
                    2€ / Price per extra per.
                  </Paragraph>
                </div>
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 210€0"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Mon-Thu
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                10€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 220€1"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Fri
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                20€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 230€2"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sat
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                30€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 240€3"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sun
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>
      </Card>
      <Card
        fluid={true}
        key="Season 32"
      >
        <CardContent>
          <Grid
            areColumnsCentered={false}
            padded={true}
          >
            <GridRow
              horizontalAlignContent="left"
            >
              <GridColumn
                verticalAlignContent="top"
                width={12}
              >
                <div>
                  <Paragraph
                    size="medium"
                    weight="heavy"
                  >
                    Season 3
                  </Paragraph>
                  <Paragraph
                    size="medium"
                    weight="light"
                  >
                    01/03/2018 - 01/08/2018
                    <br />
                    <Icon
                      color={null}
                      hasBorder={false}
                      isCircular={false}
                      isColorInverted={false}
                      isDisabled={false}
                      isLabelLeft={false}
                      labelText={null}
                      labelWeight={null}
                      name="guests"
                      path={null}
                      size={null}
                    />
                    4
                    <br />
                    3€ / Price per extra per.
                  </Paragraph>
                </div>
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 340€0"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Mon-Thu
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                40€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 330€1"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Fri
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                30€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 320€2"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sat
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                20€
              </GridColumn>
            </GridRow>
            <GridRow
              columns={2}
              horizontalAlignContent="left"
              key="Season 310€3"
            >
              <GridColumn
                floated="left"
                verticalAlignContent="top"
                width={null}
              >
                <Paragraph
                  size="medium"
                  weight="heavy"
                >
                  Sun
                </Paragraph>
              </GridColumn>
              <GridColumn
                floated="right"
                textAlign="right"
                verticalAlignContent="top"
                width={null}
              >
                10€
              </GridColumn>
            </GridRow>
          </Grid>
        </CardContent>
      </Card>
    </ShowOnMobile>
  </Grid>
  <ShowOnDesktop
    parent="div"
    parentProps={Object {}}
  >
    <Table
      className=""
      tableBody={
        Array [
          Array [
            <div>
              <Paragraph
                size="medium"
                weight="heavy"
              >
                Season 1
              </Paragraph>
              <Paragraph
                size="medium"
                weight="light"
              >
                01/05/2018 - 01/08/2018
                <br />
                <Icon
                  color={null}
                  hasBorder={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="guests"
                  path={null}
                  size={null}
                />
                2
                <br />
                1€ / Price per extra per.
              </Paragraph>
            </div>,
            "40€",
            "30€",
            "40€",
            "40€",
          ],
          Array [
            <div>
              <Paragraph
                size="medium"
                weight="heavy"
              >
                Season 2
              </Paragraph>
              <Paragraph
                size="medium"
                weight="light"
              >
                01/02/2018 - 01/08/2018
                <br />
                <Icon
                  color={null}
                  hasBorder={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="guests"
                  path={null}
                  size={null}
                />
                3
                <br />
                2€ / Price per extra per.
              </Paragraph>
            </div>,
            "10€",
            "20€",
            "30€",
            "40€",
          ],
          Array [
            <div>
              <Paragraph
                size="medium"
                weight="heavy"
              >
                Season 3
              </Paragraph>
              <Paragraph
                size="medium"
                weight="light"
              >
                01/03/2018 - 01/08/2018
                <br />
                <Icon
                  color={null}
                  hasBorder={false}
                  isCircular={false}
                  isColorInverted={false}
                  isDisabled={false}
                  isLabelLeft={false}
                  labelText={null}
                  labelWeight={null}
                  name="guests"
                  path={null}
                  size={null}
                />
                4
                <br />
                3€ / Price per extra per.
              </Paragraph>
            </div>,
            "40€",
            "30€",
            "20€",
            "10€",
          ],
        ]
      }
      tableHeadings={
        Array [
          <Dropdown
            error={false}
            icon={null}
            isDisabled={false}
            isValid={false}
            label=""
            name=""
            onBlur={[Function]}
            onChange={[Function]}
            options={
              Array [
                Object {
                  "text": "EUR €",
                  "value": "eur",
                },
                Object {
                  "text": "GBP £",
                  "value": "gbp",
                },
                Object {
                  "text": "USD $",
                  "value": "usd",
                },
              ]
            }
            willOpenAbove={false}
          />,
          "Mon-Thu",
          "Fri",
          "Sat",
          "Sun",
        ]
      }
      tableId=""
    />
  </ShowOnDesktop>
</Fragment>
`;
